packages/monaco/src/browser/monaco-semantic-highlighting-service.ts

```diff
@@ -141,3 +187,15 @@ export class MonacoSemanticHighlightingService extends SemanticHighlightingServi
     }
 
 }
+
+// TODO(nridge): Can we use this class from @theia/plugin, instead of duplicating it?
+class IdGenerator {
```

Comment:
@HighCommander4 Could you point me please where did you copy it? I could think about how we can refactor it.

Generally plugin system can use monaco extension, but not another way around. Maybe we can pull some services from the plugin system to the monaco extension to reuse.

Reply:
I copied it from [here](https://github.com/theia-ide/theia/blob/0eebcf368461d4e138befb4d07448c07f06942de/packages/plugin-ext/src/common/id-generator.ts#L17).

Reply:
ok, i think you don't need this class, you can just inline it into `MonacoSemanticHighlightingService`

Reply:
Done